launcher/src/main/java/net/consensys/cava/launcher/Launcher.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2018, ConsenSys Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package net.consensys.cava.launcher;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.io.PrintStream;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.logl.Logger;
+import org.logl.LoggerProvider;
+
+/**
+ * Provides common application bootstrapping.
+ *
+ * <p>
+ * Used to configure the behavior of the program at startup and provide handlers for command line arguments.
+ */
+public final class Launcher {
+
+  private final Map<String, LauncherHandler> handlerMap = new HashMap<>();
+
+  private LoggerProvider loggerProvider = null;
+  private LauncherHandler defaultHandler = null;
+  private PrintStream sysout = System.out;
+  private PrintStream syserr = System.err;
+
+  private LaunchExceptionHandler launchExceptionHandler = (logger, syserr, sysout, t) -> {
+    syserr.println(t.getMessage());
+    System.exit(1);
+  };
+
+  /**
+   * Runs the application with the arguments provided.
+   *
+   * @param args The command line arguments.
+   */
+  public void launch(String... args) {
+    if (loggerProvider == null) {
+      throw new IllegalStateException("Logger provider has not been set");
+    }
+    if (defaultHandler == null) {
+      throw new IllegalStateException("Default handler has not been set");
+    }
+    Logger logger = loggerProvider.getLogger("launch");
+    try {
+      for (String arg : args) {
+        LauncherHandler handler = handlerMap.get(arg);
+        if (handler != null) {
+          handler.handle(logger, sysout, syserr, args);
```

Comment:
sysout/syserr aren't used in this class - just passed directly to handle methods. Why if funnelling them via this class better than directly injecting into the handlers, or captured into the lambda closure?